app/vlagent/kubernetescollector/client.go

```diff
@@ -283,6 +283,53 @@ type nodeList struct {
 	Items []node `json:"items"`
 }
 
+type namespaceMetadata struct {
+	Name            string            `json:"name"`
+	ResourceVersion string            `json:"resourceVersion"`
+	Labels          map[string]string `json:"labels"`
+	Annotations     map[string]string `json:"annotations"`
+}
+
+type namespace struct {
+	Metadata namespaceMetadata `json:"metadata"`
```

Comment:
The `metadata` field has `objectMeta` type according to https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#namespace-v1-core . The `objectMeta` type is used across many different object types returned by Kubernetes API server, so it is better from the consistency PoV to rename the `namespaceMetadata` struct to `objectMeta`.

Reply:
Agree, I prefer the approach of adding fields only as they are needed. We shouldn't include unused fields. This is how vlagent currently works. If you find any fields that are not being used, please let me know - that would be a bug and shouldn't be there. In this PR the `namespaceMedata.ResourceVersion` field is not used and must be deleted.

While unifying the metadata structure sounds appealing, it might lead to the exact issue we are trying to avoid: bloating our objects with fields that aren't applicable to all resources. For instance, including a UID field in a shared object makes sense for Pods, but it's unnecessary for Nodes or Namespaces.

Are you sure it's a good idea to have one "catch-all" object instead of smaller, specialized objects that we can extend as needed?

Reply:
>The metadata field has objectMeta type according to https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#namespace-v1-core . The objectMeta type is used across many different object types returned by Kubernetes API server, so it is better from the consistency PoV to rename the namespaceMetadata struct to objectMeta.

done

>also makes sense to add deletionTimestamp to metadata and skip objects with non-zero values

yes, agree. already skip when namespace is  deleting

> Are you sure it's a good idea to have one "catch-all" object instead of smaller, specialized objects that we can extend as needed?

yes, fixed